src/axolotl/prompt_strategies/metharme.py

```diff
@@ -0,0 +1,77 @@
+"""Module containing the MetharmenPromptTokenizingStrategy and MetharmePrompter class"""
+
+import logging
+from typing import Tuple
+
+from axolotl.prompt_tokenizers import (
+    InstructionPromptTokenizingStrategy,
+)
+from axolotl.prompters import AlpacaPrompter
+
+LOG = logging.getLogger("axolotl")
+
+IGNORE_TOKEN_ID = -100
+
+
+class MetharmePromptTokenizingStrategy(InstructionPromptTokenizingStrategy):
+    """
+    Tokenizing strategy for the Metharme models
+    """
+
+    def parse_instruction_fields(self, prompt) -> Tuple[str, str, str]:
+        return (
+            prompt["prompt"],
+            "",
+            prompt["generation"]
+        )
+
+    def _tokenize(self, prompt: str, add_eos_token: bool = True, strip_bos_token: bool = False, num_eos_tokens: int = 3):
+        result = self.tokenizer(
+            prompt,
+            truncation=True,
+            max_length=self.sequence_len,
+            padding=False,
+            return_tensors=None,
+        )
+        if len(result["input_ids"]) == 0:
+            LOG.warning("Tokenizer result is empty. You may want to audit your dataset")
+        # If there's already an EOS token there, subtract from the number added
+        if result["input_ids"][-1] == self.tokenizer.eos_token_id:
+            num_eos_tokens -= 1
+
+        if num_eos_tokens > 0 and add_eos_token:
+            for _ in range(num_eos_tokens):
+                if (
+                    len(result["input_ids"]) > 0
+                    and len(result["input_ids"]) < self.sequence_len
+                ):
+                    result["input_ids"].append(self.tokenizer.eos_token_id)
+                    result["attention_mask"].append(1)
+
+        if result["input_ids"][0] == self.tokenizer.bos_token_id and strip_bos_token:
+            result["input_ids"] = result["input_ids"][1:]
+            result["attention_mask"] = result["attention_mask"][1:]
+
+        result["labels"] = result["input_ids"].copy()
+        return result
+
+
+class MetharmePrompter(AlpacaPrompter):
+    """
+    Prompter for the Metharme models.
+    """
+
+    system_prompt = ""
+    system_no_input_prompt = ""
+    system_format = ""
+    turn_format = "{instruction}"
+    turn_no_input_format = "{instruction}"
```

Comment:
May I ask if there's any samples of formatting you have for Metharme? I think this part might not look right. An example of alpaca is here: https://github.com/TearGosling/axolotl/blob/eba46c261c3d57e6f10dbb2b59daa1225f372781/src/axolotl/prompters.py#L22-L45

Reply:
I see what you mean. The reason we don't provide a system format here is that the system prompt is already provided within the `prompt` field. Metharme uses a ChatML-like system, like so:
```{"prompt": "<|system|>Goal: Assistant\nObjective: Follow the user's instructions<|user|>A train travels from Town A to Town B at a speed of 60 miles per hour. The return trip from Town B back to Town A is made at a speed of 40 miles per hour. What is the average speed for the entire round trip?<|model|>", "generation": "The average speed for the entire round trip is 48 miles per hour.\n\nHere's the reasoning behind this:\n\n1. To find the average speed for the entire round trip, we need to consider the harmonic mean of the two speeds, not the arithmetic mean.\n2. The formula for the harmonic mean of two numbers, x and y, is H = 2xy / (x + y).\n3. Plugging in the values for the speeds during the two trips, we get: H = 2 * 60 * 40 / (60 + 40) = 4800 / 100 = 48.\n4. Therefore, the average speed for the entire round trip is 48 miles per hour.\n\nIt's important to note that simply averaging the two speeds (60 + 40) / 2 = 50 would not give the correct answer because the time spent traveling at each speed is different due to the varying distances covered."}```

Admittedly, I am aware there is a ChatML-like system already implemented within Axolotl but due to time constraints, I think this is easier for now and works just as well.

Reply:
Additionally, this is also why we have an `__init__` function which only does a pass: the prompt is already built.

Reply:
Ok, this sounds good! Thanks for the clarification.